### DISCUSSION TITLE
Republish Github pages

### DISCUSSION THREAD

**Discussion by MrIntercept** [community/community#29619] [New to GitHub] [answered]
Hi , Im new to github... Can i republish my website if i already unpublished it? If yes how to publish it again...Thanks

**Comment by mazkdevf** [comment, marked as answer]
> Hi , Im new to github... Can i republish my website if i already unpublished it? If yes how to publish it again...Thanks

Hello,

Do you mean like you have repository what uses Github pages and you want to update something?

If so you just have to commit the new update and github scan will look that automatically 

That can take 1-10mins

**Reply by MrIntercept** [reply]
I see, thanks...

**Comment by SaadZehri2** [comment]
I account has been flagged problem please solve my problem

**Comment by IsmailUmmer** [comment]
if u lost the republish button . don't worry go to actions. click pages build and deployment , then re-run all jobs pages . your website is published again

![Screenshot 2022-09-20 222734](https://user-images.githubusercontent.com/66722869/191320916-676649d7-4498-42aa-9646-1305d0fdadcf.png)

**Reply by AGfegg** [reply]
what is that doesnt show up

**Reply by IsmailUmmer** [reply]
Which one please specify ur question clearly

**Reply by brahimgb** [reply]
![image](https://user-images.githubusercontent.com/59414164/215370823-f4d9a5d8-4f89-4e2f-baec-417b0a7c390b.png)
I unpublish my website and I want to republish it but I don't know how to do it I run `npm run deploy` but dos not work

**Reply by jmasselink** [reply]
I'm having the same issue. I had a working deployment workflow for a Github Pages site. But now, that "artifact" has expired and I can not rerun the build. I'm unsure how to create it myself
![image](https://github.com/community/community/assets/11066771/c4e138ae-69f9-4509-a2e3-6162ac4a5a14)

**Reply by Naresh-1308** [reply]
if i have unpublished the site and want to publish again the same repo then what to do

**Comment by Breitie** [comment]
Hi all, Daylinn from SOUTH AFRICA here, 
I need some help with github and i am not really getting help with google.

I have created a repository and went to pages ro activate the site that i want to upload but then it keeps giving me an error by my domain. something about the DNR not linked to my readme or just a plain error?

I dont know what i am doing wrong. will someone be able to help me?

**Reply by IsmailUmmer** [reply]
do u have a domain?

**Reply by Breitie** [reply]
yes well after going into pages. it requests for a domain which i then insert. for example 
breitie.co.za 

and then it says DNS check in progress and bamb error
![Screenshot 2022-09-20 192705](https://user-images.githubusercontent.com/112810021/191324698-e622abc8-7ff9-4c39-8fa5-9b10e21da325.jpg)

**Reply by IsmailUmmer** [reply]
go to this page https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site 

read the 5 th point 
copy the ips and go to your purchased domain site and right click on your domain and click manage dns 
edit the A row . and change points to will be parked change it to that ip on documentation one by one (example:185.199.108.153) then save it 
then add an another A tag , host will be @ and points to next ip 
once all ip is complted then come back and save

**Reply by Breitie** [reply]
Hi sorry to be only replying now. hectic week. 
did I have to purchase a domain? I do not know how to access my DNS provider, didnt even know i had one and then it says i should configure the domain to link to my username.github.io name? I am really lost and struggeling big time. I have deleted and resubmitted my repository doing everything that the troubleshoot says to but i just keep hitting the DNS error.

**Reply by airtower-luna** [reply]
> did I have to purchase a domain? I do not know how to access my DNS provider, didnt even know i had one

Yes. And if you don't know about a DNS provider ("[domain name registrar](https://en.wikipedia.org/wiki/Domain_name_registrar)"), you probably don't have one yet. You only need one if you register a domain.

So you need to find a registrar that offers `co.za` domains and buy the one you want. Or you can use the `github.io` domain for your username.

**Reply by OuaoulmestOttman** [reply]
to rebuild the website all u need to do is change main to none on settings/pages/branch then put it back to main.

**Comment by Rod-Freedom** [comment]
> Hi , Im new to github... Can i republish my website if i already unpublished it? If yes how to publish it again...Thanks

I'm late as hell, but the issue is that you're looking for the option in the "Actions" menu, but your answer is in the repo's settings menu; then look for the "pages" option under "Code and automation", and there, you'll find the option to deploy it.
